### PR TITLE
Add support for loading "trackless" SLEAP files

### LINKED ISSUE #89: Support reading SLEAP files without Tracks
When reading SLEAP files (.slp and .h5), _movement_ currently [expects to find at least one SLEAP Track](https://github.com/neuroinformatics-unit/movement/blob/7e12f08b49dcab6ee3924861ad7ae6cfa3be172c/movement/io/load_poses.py#L118-L119). However, when one uses [`sleap-track`](https://sleap.ai/guides/cli.html#inference-and-tracking) for running inference in SLEAP,  `--tracking.tracker` is set to `None` by default, i.e., the resulting .slp output file for a single-animal dataset would have 0 tracks: `Labels(labeled_frames=18296, videos=1, skeletons=1, tracks=0)` and  `f["track_names"][:]` would be empty in the corresponding .h5 output. 

As a result, `load_poses.from_sleap_file("file.h5")` would fail, while `load_poses.from_sleap_file("file.slp")` would return a dataset with 0 individuals and pose_tracks with `shape=(18485, 0, 6, 2)`.

**Possible solution**
- Provide default `individual_names` if `track` is `None`, when reading SLEAP [.h5 analysis files](https://github.com/neuroinformatics-unit/movement/blob/7e12f08b49dcab6ee3924861ad7ae6cfa3be172c/movement/io/load_poses.py#L248) and [.slp files](https://github.com/neuroinformatics-unit/movement/blob/7e12f08b49dcab6ee3924861ad7ae6cfa3be172c/movement/io/load_poses.py#L280)
- Log warning when setting default `individual_names`

### PR DESCRIPTION
**Pull request by lochhh**
## Description

**What is this PR**

- [ ] Bug fix
- [x] Addition of a new feature
- [ ] Other

**Why is this PR needed?**
This PR enables SLEAP files (.h5 and .slp) without Tracks to be loaded.

**What does this PR do?**

## References
This PR resolves #89 

## How has this PR been tested?
Relevant tests have been added and run

## Checklist:

- [x] The code has been tested locally
- [x] Tests have been added to cover all new functionality
- [ ] The documentation has been updated to reflect any changes
- [x] The code has been formatted with [pre-commit](https://pre-commit.com/)

### CODE REVIEW

File: movement/io/load_poses.py [hunk 1]
```diff
@@ -238,14 +242,21 @@ def _load_from_sleap_analysis_file(
         tracks = f["tracks"][:].transpose((3, 0, 2, 1))
         # Create an array of NaNs for the confidence scores
         scores = np.full(tracks.shape[:-1], np.nan)
+        individual_names = [n.decode() for n in f["track_names"][:]] or None
+        if individual_names is None:
+            log_warning(
+                f"Could not find SLEAP Track in {file.path}. "
+                "Assuming single-individual dataset and assigning "
+                "default individual name."
```
Comment (niksirbi) [inline movement/io/load_poses.py:250]: Is it worth mentioning here what that default individual name is?

File: movement/io/load_poses.py [hunk 2]
```diff
@@ -274,10 +285,17 @@ def _load_from_sleap_labels_file(
     file = ValidHDF5(file_path, expected_datasets=["pred_points", "metadata"])
     labels = read_labels(file.path.as_posix())
     tracks_with_scores = _sleap_labels_to_numpy(labels)
+    individual_names = [track.name for track in labels.tracks] or None
+    if individual_names is None:
+        log_warning(
+            f"Could not find SLEAP Track in {file.path}. "
+            "Assuming single-individual dataset and assigning "
+            "default individual name."
```
Comment (niksirbi) [inline movement/io/load_poses.py:293]: also consider mentionin what the default name is here

Comment (niksirbi) [reply movement/io/load_poses.py:293]: i.e. "individual_0"

**Review by niksirbi — APPROVED** [review]
LGTM! I also tested in on the sample data for the course, and it works!

File: movement/io/load_poses.py (hunk 2 above)
Comment (lochhh) [reply movement/io/load_poses.py:293]: I would say we don't, for maintainability, plus [`ValidPoseTracks`](https://github.com/neuroinformatics-unit/movement/blob/7e12f08b49dcab6ee3924861ad7ae6cfa3be172c/movement/io/validators.py#L346-L349) will tell what names are set

Comment (niksirbi) [reply movement/io/load_poses.py:293]: I'm fine with that.

**Pull request merged**